mapclientplugins/geometricfitstep/view/geometricfitwidget.py

```diff
@@ -528,6 +545,76 @@ def _configProjectionCentreGroupsClicked(self):
                 self._refreshStepItem(config)
                 self._refreshGraphics()
+    
+    def _configSettingGroupChanged(self, index):
+        """
+        Callback for change in setting group field chooser widget.
+        """
+        self._updateDataProportion()
+
+    def _updateDataProportion(self):
+        realFormat = "{:.4g}"
+        lineEditDisable = True
+        checkBoxDisable = True
+        checkBoxTristate = False
+        dataProportionStr = "1.0"
+        checkBoxState = QtCore.Qt.Unchecked
+        config = self._getConfig()
+        group = self._ui.configSettingGroup_fieldChooser.getField()
+        if group:
+            checkBoxDisable = False
+            proportion, isLocallySet = config.getGroupDataProportion(group.getName())
+            #print("group:", group.getName(), "proportion", proportion, isLocallySet)
+            if proportion:
+                dataProportionStr = realFormat.format(proportion)
+                if isLocallySet:
+                    checkBoxState = QtCore.Qt.Checked
+                    lineEditDisable = False
+                else:      
+                    checkBoxState = QtCore.Qt.PartiallyChecked
+                    checkBoxTristate = True
+            #elif isLocallySet:
+                #checkBoxTristate = True
+        self._ui.configDataProportion_checkBox.setDisabled(checkBoxDisable)
+        self._ui.configDataProportion_checkBox.setCheckState(checkBoxState)
+        self._ui.configDataProportion_checkBox.setTristate(checkBoxTristate)
+        self._ui.configDataProportion_lineEdit.setDisabled(lineEditDisable)
+        self._ui.configDataProportion_lineEdit.setText(dataProportionStr)
+    
+    def _configDataProportionClicked(self):
+        checkState = self._ui.configDataProportion_checkBox.checkState()
+        triState = 0 if (checkState == QtCore.Qt.Unchecked) else 1 if (checkState == QtCore.Qt.PartiallyChecked) else 2
+        group = self._ui.configSettingGroup_fieldChooser.getField().getName()
+        if triState == 2:
+            self._ui.configDataProportion_lineEdit.setDisabled(False)
+        else:
+            self._ui.configDataProportion_lineEdit.setDisabled(True)
+        if triState == 0:
+            self._getConfig().setGroupDataProportion(group, None)  
+
+    def _configDataProportionEntered(self):
+        value = QLineEdit_parseRealNonNegative(self._ui.configDataProportion_lineEdit)
+        group = self._ui.configSettingGroup_fieldChooser.getField().getName()
+        if value > 0.0:
+            self._getConfig().setGroupDataProportion(group, value)
+        else:
+            print("Invalid model Data Proportion entered")
+        self._updateConfigWidgets()
+
+    def _configDataWeightClicked(self):
+        checkState = self._ui.configDataWeight_checkBox.checkState()
+        triState = 0 if (checkState == QtCore.Qt.Unchecked) else 1 if (checkState == QtCore.Qt.PartiallyChecked) else 2
+        self._fitter.setConfigDataWeight(triState)
```

Comment:
I don't believe this API is ready yet -- the widget is hidden so it should never be called. Perhaps I can get it working today and we'll get both bits in the PR.